nesta/packages/nlp_utils/ngrammer.py

```diff
@@ -94,7 +94,9 @@ def process_document(self, raw_text, remove_stops=True):
             processed_doc = []
             for sentence in text:
                 sentence = [token for token in sentence
-                            if token not in stop_words]
+                            if token not in stop_words and
+                            not all(t in stop_words
+                                    for t in token.split('_'))]
```

Comment:
What happens with garbage ngrams such as `of_technology` and `the_machine_learning`? Would it be helpful to keep the stop word only when `len(token.split('_')) > 2` or when it is at the middle of the `token.split('_')` list?

Reply:
the assumption is that the number of garbage n-grams is small in the wiktionary dataset. I think this is a limitation at the moment that I can't think of a robust answer for, since terms like `of_biblical_proportions` (actually in the dataset) are relevant, whereas `of_an`  (also in the dataset) are irrelevant, and at least the above procedure works for these examples.